sip_assembly/views.py

```diff
@@ -32,19 +32,38 @@ def get_serializer_class(self):
         return SIPSerializer
 
     def create(self, request):
+        """
+        Allow for post requests from Ursa Major 0.x or 1.x
+        """
         try:
-            sip = SIP(
-                process_status=10,
-                bag_path=join(
-                    settings.BASE_DIR,
-                    settings.SRC_DIR,
-                    "{}.tar.gz".format(
-                        request.data['identifier'])),
-                bag_identifier=request.data['identifier'],
-                data=request.data['bag_data'],
-                origin=request.data['origin']
-            )
-            sip.save()
+            if request.data.get('bag_data'):
+                sip = SIP(
+                    process_status=10,
+                    bag_path=join(
+                        settings.BASE_DIR,
+                        settings.SRC_DIR,
+                        "{}.tar.gz".format(
+                            request.data['identifier'])),
+                    bag_identifier=request.data['identifier'],
+                    data=request.data['bag_data'],
+                    origin=request.data['origin']
+                )
+                sip.save()
+                if request.data.get('origin'):
+                    sip.origin = request.data.get('origin')
+                    sip.save()
```

Comment:
I think we can presume that this key will not be null if the data matches this first case.